src/lib/device-logs/backends/redis.ts

```diff
@@ -76,19 +76,25 @@ export class RedisBackend implements DeviceLogsBackend {
 		const redisLogs = logs.map(this.toRedisLog, this);
 
 		const limit = ctx.retention_limit || 0;
-		const key = this.getKey(ctx);
+		const logKey = this.getKey(ctx);
+		const logBytesWrittenKey = this.getLogBytesWrittenKey(ctx);
 		// Create a Redis transaction
 		const tx = this.cmds.multi();
 		// Add the logs to the List structure
-		tx.rpush(key, redisLogs);
+		tx.rpush(logKey, redisLogs);
 		// Trim it to the retention limit
-		tx.ltrim(key, -limit, -1);
+		tx.ltrim(logKey, -limit, -1);
 		// Publish each log using Redis PubSub
+		let logBytesWritten = 0;
 		for (const rLog of redisLogs) {
-			tx.publish(key, rLog);
+			tx.publish(logKey, rLog);
+			logBytesWritten += rLog.length;
```

Comment:
This specifically is the size of the avsc serialized binary data which is fairly transformed vs the actual logs we receive - that may be fine and will still give an idea of the relative log usage between devices but it should be used as guidance rather than precise figures.

Reply:
@Page- is there an `O(1)` storage complexity difference between serialized and raw logs received?

Reply:
And if so, if that constant factor is small enough, it may be alright to just look at serialized size. But I think we are also interested in the actual *network transfer bytes*, represented by the data seen at `POST /device_logs`, vs seen here which is more representative of *bytes written to disk*.

Reply:
@petrosagg @Page- brings a good point here. 

Given we are interested in this data from a capacity planning perspective, we might want metrics on both the serialized and raw logs (the former represents storage space, the latter network transfer). 

For the former (serialized logs sent to redis), this PR suffices. For the latter, (raw logs bytes seen from devices at the soonest ingestion point, `POST /device_logs`) I can easily add `redis.incrby()` calls to the `/device_logs` endpoint as well. 

My question is thus, summarized: would we like both (given that it's trivial to implement both, and then scan both in our exporter and have two histograms). There may be some useful difference to tracking both. Given the work done so far, this would be like a ~10 line change to both the API and the exporter to add that ability.

Reply:
In fact I'm working this out as a PR now.

Reply:
In the branch I'm working out now (soon posted), we will track:

- `device:*:logBytesWritten` for serialized size in redis keystore
- `device:*:logBytesReceived` for bytes received at the `/device_logs` endpoint

Reply:
In researching this, I've found even this PR doesn't handle streaming writes. My first order of business is to make sure this PR includes that, as we want at least an accurate measure of bytes written to REDIS.

Reply:
This PR at any rate, to keep track of bytes written, is ready to go. I had a slight worry that it didn't handle streaming writes, but since the instrumentation is added in the `publish()` method, which is used by both `/device/v2/:uuid/logs` and `/device/v2/:uuid/log-stream` (in the `handleStreamingWrite()` method) it's good, this will track both regular and streaming logs bytes written.